js/copycode.js

```diff
@@ -13,6 +14,15 @@ codeBlocks.forEach((block) => {
   });
 });
 
+// add heading links
+for (const heading of headers) {

```

Comment:
```suggestion
for (const heading of headers) {
    if (heading.querySelector('.heading-link')) continue;
```
Nit: A little paranoid but this will guard against the script running twice somehow by accident or hot reloading or something, preventing us from adding the elements twice.

Reply:
I want thinking of implementing lazy attach links using  `IntersectionObserver` or `MutationObserver`. WDYT?

> Is there a reason why this is living inside the copycode.js script file?

Copy code and header link share same logic and needs asynchronous implementation.

Reply:
The DOM is reloaded completely on each hot reload, so duplicates won’t occur. Still adding this check as a defensive measure in case the script ever re-runs